src/components/structures/UserMenu.tsx

```diff
@@ -60,31 +68,80 @@ const CustomStatusSection = () => {
     const cli = useContext(MatrixClientContext);
     const setStatus = cli.getUser(cli.getUserId()).unstable_statusMessage || "";
     const [value, setValue] = useState(setStatus);
+    const [isFocused, setFocus] = useState(false);
+
+    const ref = useRef<HTMLInputElement>(null);
+    const [rovingOnFocus, isActive] = useRovingTabIndex(ref);
+    const rovingContext = useContext(RovingTabIndexContext);
+
+    const classes = classNames({
+        'mx_UserMenu_CustomStatusSection_field': true,
+        'mx_UserMenu_CustomStatusSection_field_hasQuery': value,
+        'mx_UserMenu_CustomStatusSection_field_focused': isFocused,
+    });
+
+    const onFormKeyDown = (ev: React.KeyboardEvent) => {
+        // Allow the custom status form to be tabbed through to move from the
+        // <input> to the buttons and back to the normal options. Normally,
+        // RoamingTabIndex items can only be navigated by Arrow keys but we need
+        // to be able to jump the focus out of the <input> so we also allow Tab
+        // within the form.
+        if (ev.key === Key.TAB && rovingContext.state.refs.length > 0) {
```

Comment:
wonder if we should/need to check for cmd/ctrl here, not sure if browsers let webapps override tab change keybinds.

Reply:
😬 I was also thinking about this too. I'm sure people have their own keybindings for moving focus. But we already had assumptions about `Tab` so I went with it.

Wish there was a way to detect focus moved key binding.

Reply:
I mean so we don't trigger this on <kbd>Ctrl/Cmd/Super/Alt</kbd> + <kbd>Tab</kbd>

Reply:
On macOS, you can press `Apple + Tab` to switch apps. When I'm in the input, pressing that shortcut does the proper OS action and isn't consumed by our app. I don't think we need to worry about it.

I could see someone having a screenreader shortcut that might interfere since it integrates into the browser itself instead of the OS.

We don't do this for any of our other shortcuts atm so I am leaning to not worrying about it yet until we have someone explaining a specific scenario to mitigate for.